src/capymoa/regressor/_soknl.py

```diff
@@ -1,17 +1,40 @@
 # Library imports
-from typing import Optional, Union
+
+from ._soknl_base_tree import SOKNLBT
+from moa.classifiers.meta import SelfOptimisingKNearestLeaves as _MOA_SOKNL
 
 from capymoa.base import (
     MOARegressor, _extract_moa_learner_CLI
 )
 
-from capymoa.splitcriteria import SplitCriterion, _split_criterion_to_cli_str
-from capymoa.stream._stream import Schema
-from moa.classifiers.meta import SelfOptimisingKNearestLeaves as _MOA_SOKNL
-from ._soknl_base_tree import SOKNLBT
-
 
 class SOKNL(MOARegressor):
+    """Self-Optimising K-Nearest Leaves (SOKNL) Implementation.
+
+    SOKNL is an extension to AdaptiveRandomForestRegressor.
+    SOKNL stores abstract information for all instance seen by the leaves, use them to calculate the distance
```

Comment:
Improve this part. What is the abstract information? What is the certain instance? You may refer to it as the instance whose target value is being predicted (or find a more concise way to refer to it :) ). Also, you are not explaining the Self-Optimising mechanism in here, and you should.